packages/wrangler/src/index.tsx

```diff
@@ -1165,6 +1165,97 @@ function createCLIParser(argv: string[]) {
 		}
 	);
 
+	wrangler.command(
+		"secret:bulk [json]",
+		"🗄️ Bulk upload secrets for a Worker",
+		(yargs) => {
+			return yargs
+				.positional("json", {
+					describe: `The JSON file of key-value pairs to upload, in form {"key": value, ...}`,
+					type: "string",
+				})
+				.option("name", {
+					describe: "Name of the Worker",
+					type: "string",
+					requiresArg: true,
+				})
+				.option("env", {
+					type: "string",
+					requiresArg: true,
+					describe:
+						"Binds the secret to the Worker of the specific environment.",
+					alias: "e",
+				});
+		},
+		async (secretBulkArgs) => {
+			await printWranglerBanner();
+			const config = readConfig(
+				secretBulkArgs.config as ConfigPath,
+				secretBulkArgs
+			);
+
+			const scriptName = getLegacyScriptName(secretBulkArgs, config);
+			if (!scriptName) {
+				throw new Error(
+					"Required Worker name missing. Please specify the Worker name in wrangler.toml, or pass it as an argument with `--name <worker-name>`"
+				);
+			}
+
+			const accountId = await requireAuth(config);
+
+			if (secretBulkArgs.json) {
+				logger.log(
+					`🌀 Creating the secrets for the Worker "${scriptName}" ${
+						secretBulkArgs.env && !isLegacyEnv(config)
+							? `(${secretBulkArgs.env})`
+							: ""
+					}`
+				);
+				const jsonFilePath = path.resolve(secretBulkArgs.json);
+				const content = parseJSON<Record<string, string>>(
+					readFileSync(jsonFilePath),
+					jsonFilePath
+				);
+				for (const key in content) {
+					if (typeof content[key] !== "string") {
+						throw new Error(
+							`The value for ${key} in ${jsonFilePath} is not a string.`
+						);
+					}
+				}
+
+				const url =
+					!secretBulkArgs.env || isLegacyEnv(config)
+						? `/accounts/${accountId}/workers/scripts/${scriptName}/secrets`
+						: `/accounts/${accountId}/workers/services/${scriptName}/environments/${secretBulkArgs.env}/secrets`;
+				// Until we have a bulk route for secrets, we need to make a request for each key/value pair
+				await Promise.allSettled(
+					Object.entries(content).map(async ([key, value]) => {
+						return fetchResult(url, {
+							method: "PUT",
+							headers: { "Content-Type": "application/json" },
+							body: JSON.stringify({
+								name: key,
+								text: value,
+								type: "secret_text",
+							}),
+						})
+							.then(() => {
+								logger.log(`✨ Successfully created secret for key: ${key}`);
+							})
+							.catch((e) => {
+								logger.error(
+									`🚨 Error uploading secret for key: ${key}:
+										${e.message}`
+								);
+							});
+					})
+				);
+				return logger.log("✨ Successfully created secrets");
```

Comment:
This isn't strictly true as some might not have been successfully created. Maybe we should add a summary for `x` successfully created and `y` not uploaded - we get this information from Promise.allSettled.

My suggestion would be an output like:
```
✨ Successfully created secret for key: secret-name-1
...
🚨 Error uploading secret for key: secret-name-8
✨ Successfully created secret for key: secret-name-9

Summary:
✨ 7 secrets successfully uploaded
🚨 2 secrets failed to upload
```
and to only display the line with number of failures if there were any failures

Reply:
This can be additional scope, I suggest making an issue for polish/enhancement.

Reply:
Even without the extra scope of the summary section, I still think we shouldn't be logging `Successfully created secrets` if some (or all) have failed to be created

Reply:
> I still think we shouldn't be logging Successfully created secrets if some (or all) have failed to be created.

I agree.